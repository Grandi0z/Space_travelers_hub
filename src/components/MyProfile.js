import React from 'react';
import Table from 'react-bootstrap/Table';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const { missions } = useSelector((store) => store.missions);
  const selectJoinedMission = missions.filter((mission) => mission.joined);
  return (
    <div>
      <>
        <h2>My Missions</h2>
        {selectJoinedMission.length && (
        <Table striped bordered="true" hover>
          <tbody>
            {selectJoinedMission.map((mission) => (
              <tr key={mission.mission_id}>
                <td>{mission.mission_name}</td>
              </tr>
            ))}
          </tbody>
        </Table>
        )}

      </>
    </div>
  );
};

export default MyProfile;
