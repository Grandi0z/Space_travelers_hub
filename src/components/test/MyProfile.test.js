import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import MyProfile from '../MyProfile';
import missionsReducer from '../../redux/features/mission/missionsSlice';
import rocketsReducer from '../../redux/features/rockets/reducers';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketsReducer,
  },
});

describe('MyProfile', () => {
  it('should render a table with my rockets', () => {
    const tree = renderer.create(
      <Provider store={store}><MyProfile /></Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
