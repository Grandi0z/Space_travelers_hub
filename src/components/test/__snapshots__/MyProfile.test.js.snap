// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile should render a table with my rockets 1`] = `
<div
  className="my-profile"
>
  <div
    className="my-profile-item"
  >
    <h2>
      My Rockets
    </h2>
    0
  </div>
  <div
    className="my-profile-item"
  >
    <h2>
      My Missions
    </h2>
    0
  </div>
</div>
`;
